srcdeps-core/src/main/java/org/l2x6/srcdeps/core/fs/PathLock.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright 2015-2016 Maven Source Dependencies
+ * Plugin contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.l2x6.srcdeps.core.fs;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.nio.file.Path;
+import java.util.concurrent.locks.ReentrantLock;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A lock which guarantees its holder to have an exclusive access to {@link #getPath()}.
+ *
+ * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
+ */
+public class PathLock implements Closeable {
+    private static final Logger log = LoggerFactory.getLogger(PathLock.class);
+    private final Path lockFilePath;
+    private final Path path;
+    private final RandomAccessFile lockFile;
+    private final ReentrantLock threadLevelLock;
+
+    public PathLock(Path path, RandomAccessFile lockFile, Path lockFilePath, ReentrantLock threadLevelLock) {
+        this.path = path;
+        this.lockFile = lockFile;
+        this.lockFilePath = lockFilePath;
+        this.threadLevelLock = threadLevelLock;
+    }
+
+    /**
+     * @return the {@link Path} locked
+     */
+    public Path getPath() {
+        return path;
+    }
+
+    /**
+     * Releases this {@link PathLock}.
+     */
+    @Override
+    public void close() {
+        try {
+            lockFile.close();
+        } catch (IOException e) {
+            log.warn(String.format("Could not close lock file [%s]", lockFilePath), e);
+        }
+        threadLevelLock.unlock();
```

Comment:
> then I would expect constructor/static method of this class to be used to obtain a lock, a-la: try (new PathLock(myPath))

I do not think I can make it work like `try (new PathLock(myPath))` because I need a central place in my app where the map from paths to the `ReentrantLock` s is maintained. `PathLocker.locks` serves exactly this purpose. 

Maybe you see a way how to lock a FS path without needing such an app wide singleton map? Sure a static locks map in `PathLock` would serve the purpose, but `srcdeps-core` is supposed to be used inside JSR-330 CDI apps and having an explict `PathLocker` `@Singleton` just suits the CDI pattern better.